### ISSUE TITLE
addinv: null obj after quiver merge otyp=444

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#410]
```
(rr) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=139749228627904) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=139749228627904) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=139749228627904, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007f19e6dc0476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007f19e6da67f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x0000559ac466ebd3 in NH_abort () at end.c:236
#6  0x0000559ac4675c3b in panic (str=0x559ac516cec0 "addinv: null obj after quiver merge otyp=%d") at end.c:802
#7  0x0000559ac47288e9 in addinv_core0 (obj=0x0, other_obj=0x0, update_perm_invent=1 '\001') at invent.c:975
#8  0x0000559ac4728fd4 in addinv (obj=0x60c00018d240) at invent.c:1026
#9  0x0000559ac439fe25 in light_cocktail (optr=0x7fff87f723b0) at apply.c:1680
#10 0x0000559ac43d1e25 in doapply () at apply.c:5103
#11 0x0000559ac448364a in rhack (cmd=0x559ac549e1e0 <in_line> "a") at cmd.c:5774
#12 0x0000559ac4376f29 in moveloop (resuming=0 '\000') at allmain.c:866
#13 0x0000559ac5068f27 in main (argc=0, argv=0x7fff87f72798) at ../sys/unix/unixmain.c:353

#define	POT_OIL	444

(rr) p moves
$1 = 23925199
```

**Comment by elunna** [comment]
The rr recording doesn't work anymore (due to kernel updates). So we'll try reproducing:

Wow.. Actually this was surprisingly easy to reproduce (or I got very lucky).

- Wish for a stack of 10 oil potions.
- quiver the stack
- apply an oil potion to light it.
- wield the lit potion
- apply the potion again to snuff it.
- Oops...

```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7360859 in __GI_abort () at abort.c:79
#2  0x0000555555a90044 in NH_abort () at end.c:236
#3  0x0000555555a9714d in panic (
    str=0x5555565a8000 "addinv: null obj after quiver merge otyp=%d") at end.c:803
#4  0x0000555555b4d0fe in addinv_core0 (obj=0x0, other_obj=0x0, update_perm_invent=1 '\001')
    at invent.c:984
#5  0x0000555555b4d7e7 in addinv (obj=0x60c000001e40) at invent.c:1035
#6  0x00005555557babe9 in light_cocktail (optr=0x7fffffffe260) at apply.c:1680
#7  0x00005555557ecad0 in doapply () at apply.c:5147
#8  0x000055555589e39b in rhack (cmd=0x5555568e1760 <in_line> "a") at cmd.c:5779
#9  0x0000555555791f94 in moveloop (resuming=0 '\000') at allmain.c:873
#10 0x00005555564a4205 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
```

**Comment by elunna** [comment]
Fixed in 3faa0029f
